src/main/java/com/musala/atmosphere/commons/websocket/message/RequestMessage.java

```diff
@@ -0,0 +1,81 @@
+package com.musala.atmosphere.commons.websocket.message;
+
+import com.musala.atmosphere.commons.RoutingAction;
+
+/**
+ * Represents a common class for all WebSocket request actions.
+ *
+ * @author dimcho.nedev
+ *
+ */
+public class RequestMessage extends Message {
+    private Object[] arguments;
+
+    private String[] typeNames;
+
+    private String deviceId;
+
+    public RequestMessage(MessageAction messageAction, RoutingAction routingAction, Object... arguments) {
+        super(messageAction, routingAction);
+        this.arguments = arguments;
+        typeNames = new String[arguments.length];
```

Comment:
Please add a comment, why do you need these names. The class by itself has no logic around them.